### PR TITLE
feat: Autosplitting

### LINKED ISSUE #4: Async/Await?
Need to check/make plugin work with async await syntax

### PR DESCRIPTION
**Pull request by ScriptedAlchemy**
Use and polyfill native import() 
Removed Babel plugin dependence. 

fix #4 and #21

### CODE REVIEW

File: src/webpack/index.js [hunk 1]
```diff
@@ -74,9 +159,46 @@ class URLImportPlugin {
           name = path;
         }
 
+        // console.log('stats', stats);
+
+        if (externalModules[chunk.id]) {
```
Comment (ScriptedAlchemy) [inline src/webpack/index.js]: @theKashey pretty much this loop. Itll tell you about the dependencies on each module. 
`const module of dependencyModuleSet.chunksIterable` lets you loop over the modules in the dependency set. I use (or will be) to get `module.chunks.foreach((chunk)=>chunk.files)`

Comment (ScriptedAlchemy) [reply src/webpack/index.js]: There's probably a lower level hook you would tap, but inside those chunks are the orderings they need to be loaded in. If a dependency matrix was created, you be able to look up any order

File: src/webpack/index.js [hunk 2]
```diff
@@ -74,9 +159,46 @@ class URLImportPlugin {
...
+          // TODO: swap forEachModle out with const of
+          // const module of chunk.modulesIterable
+          chunk.forEachModule((module) => {
+            if (module.dependencies) {
+              module.dependencies.forEach((dependency) => {
```
Comment (ScriptedAlchemy) [inline src/webpack/index.js]: @mhink Looping dependencies per module

File: src/webpack/index.js [hunk 3]
```diff
@@ -39,8 +67,60 @@ class URLImportPlugin {
   }
 
   apply(compiler) {
-    const moduleAssets = {};
+    const options = compiler?.options;
+    const chunkSplitting = options?.optimization?.splitChunks?.cacheGroups || {};
+    chunkSplitting.interleave = {
+      test(module) {
```
Comment (ScriptedAlchemy) [inline src/webpack/index.js]: @theKashey this was pretty neat. `test` provides the whole module, so im able to actually look up the module source and see if its an externalized chunk before placing it into a cacheGroup, i then can give the files a name. This allows me to code split anything by just moving them into another cache group using splitChunks and testing that the module meets the requirement... Automated code-splitting regardless of sync or async. Need to see how it holds up against tree shaking

**Comment by theKashey** [comment]
🎉🎉🎉🎉

**Comment by ScriptedAlchemy** [comment]
@theKashey i discovered some new things that I’ll put up for PR soon. You’ll like

**Pull request merged**